Source/WebCore/html/parser/HTMLConstructionSite.cpp

```diff
@@ -568,9 +569,11 @@ void HTMLConstructionSite::insertHTMLTemplateElement(AtomHTMLToken&& token)
                 clonable = ShadowRootClonable::Yes;
             else if (attribute.name() == HTMLNames::shadowrootserializableAttr)
                 serializable = ShadowRootSerializable::Yes;
+            else if (document().settings().shadowRootReferenceTargetEnabled() && attribute.name() == HTMLNames::shadowrootreferencetargetAttr)
```

Comment:
Is a settings check superfluous here?

Reply:
This feels like a good one to keep to me, but I'm not as familiar with this part of the codebase, so might be a good question to carry forward.